docs/enhancements/on-prem-data-collection.md

```diff
@@ -0,0 +1,121 @@
+
+
+## Proposal
+
+A new flow for sending data is to be added:
+When a cluster installation attempt finishes (regardless of success or failure), all of the event data for this cluster will be queried from the DB, packaged, and sent to the data hosting endpoint.
+
+Changes that need to be made in the `Assisted-Service` to support this:
+- Needs to know what the data hosting service endpoint is and how to authenticate into it
+- Needs code changes when an install is attempted to query data
+- Needs to know how to format the data to send to the data hosting service
+- Needs to expose an option for users to opt out of this feature
+
+Additional changes needed:
+- Documentation regarding terms and conditions?
+- Documentation about how to opt out
+- Documentation of requirements: non-disconnected environment
+
+Sending data should not interfere with customer workloads and deployment
+
+### User Stories
+
+#### Story 1
+
+As an infrastructure operator, I can provision and manage my fleet of clusters knowing that data points are automatically collected and sent to the Assisted Installer team without having to do additional configurations. I know my data will be safe and secure and the team will only collect data they need to improve the product. 
+
+
+#### Story 2
+
+As a developer on the assisted installer team, I can analyze the customer data to determine if a feature is worth implementing/keeping/improving. I know that the customer data is accurate and up-to-date. All of the data is parse-able and can be easily tailored to the graphs/visualizations that help my analysis.
+
+#### Story 3
+
+As a product owner, I can determine if the product is moving in the right direction based on the actual customer data. I can help determine which features and bug fixes to prioritize based on the data.
+
+### Implementation Details/Notes/Constraints
+
+Assisted Service changes:
+1. New environment variables to be added:
+    - `DISABLE_DATA_COLLECTION`: If set to `true`, it will disable this feature and thereby allow a customer to opt-out, otherwise the feature is on by default.
+    - `DATA_COLLECTION_URL`: The URL the data will be sent to.
+2. Add additional logic when a cluster reaches one of the following end stages: `error`, `cancelled`, `installed`
+    - Each of these cluster stages represent an end state and that the cluster attempted installation.
+    - Additional logic includes: querying the DB for events of that cluster and streaming this event data to the `DATA_COLLECTION_URL` over HTTPS.
+        - If the customer has large-scale clusters, would it be better to always compress the data in a separate thread?
+3. Add logic for authenticating to the `DATA_COLLECTION_URL` 
+    - Currently unknown since the service to host this data hasn't been decided on yet, but it could take the form of using a Kubernetes Secret
+
+### Risks and Mitigations
```

Comment:
what will happen if we will change the cluster/ host/ event data structure and the on-prem environment will contain old code? we need to think about backward support in our data flow

Reply:
It's a good point, but I think data format issues should be handled by services that will process the data, I think we can ignore shcema when sending it

Reply:
Good point and agreed with Riccardo. I'll add that to the non-goals section

Reply:
Do you think it would be a good idea to add a `version` field to the data? It could be the assisted-service version. That way when it reaches the service that's processing the data, it'll know what fields to expect?